samples/nrf9160/lwm2m_carrier/prj.conf

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+CONFIG_LWM2M_CARRIER=y
+
+# LWM2M Carrier requires newlib
+CONFIG_NEWLIB_LIBC=y
```

Comment:
Me and Robert have discussed this and basically there are many limitations to `select` in Kconfig, i.e. we would only be able to turn on a setting. It would be impossible to turn something off or change a numerical/string value. Thus, it might be better not to scatter the required Kconfig settings between Kconfig/`prj.conf` and keep them in one place instead.

Reply:
That is not a bad idea, it is possible to `depend on` all these settings indeed.
However even if all dependencies were in Kconfig, turning `LWM2M_CARRIER` on won't select any of them; we would still have to provide a `prj.conf` file that shows what the library needs, and then that information would be duplicated in both `prj.conf` and Kconfig. It is true that having the dependencies in Kconfig prevents the user from configuring it incorrectly but the price is having to duplicate the list of dependencies. I don't see a really big problem with that, just a very big Kconfig file :)

Reply:
This price is acceptable IMHO, assuming the user is supposed to be able to add the LWM2M carrier to his own application and not just use the lwm2m_carrier app.

Reply:
Added dependencies to Kconfig.